sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/SetFns.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkNotNull;
+
+import org.apache.beam.sdk.transforms.join.CoGbkResult;
+import org.apache.beam.sdk.transforms.join.CoGroupByKey;
+import org.apache.beam.sdk.transforms.join.KeyedPCollectionTuple;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Iterables;
+
+public class SetFns {
```

Comment:
Though not as chain-able, a `PTransform<PCollectionList<T>, PCollection<T>>` feels more natural for many of these operations.

Reply:
I feel as you told it is not chain-able, keep it `PTransform<PCollection<T>, PCollection<T>>`. Most of the time which I encountered, you will have to use a different set operation followed by another operation. like union and then Intersect or so.

Reply:
Depends on if we treat SET as binary operation or a "set" operation. If it is binary, this PR's implementation makes sense (plus it is chain-able). If we consider it is "set" operation, then it should be `PTransform<PCollectionList<T>, PCollection<T>>` by the nature of a set.

Reply:
Thanks. I understand the reasoning. My thinking was in lines similar like a Join implementation. As an end-user abstraction, isn't it would be more natural and straightforward to think left diff right which abstract that there is CoGrouping inside. This looks in my mind a natural usecase. 

Or will it be good idea to both variants `PCollectionList<T>` and `PCollection<T>` of API?

Reply:
Hi @amaliujia I tried to implement both APIs and it looks like following:
1. Binary version: `left.apply("strings", SetFns.intersect(right))`
2. PCollectionList version: `PCollectionList.of(set1).and(set1).and(set3).apply("stringsColls", SetFns.intersect())` will return. (set1 intersection set2) intersection set3 results.
Please give your thoughts.